ebean-test/src/test/java/org/tests/cascade/TestOrderedList.java

```diff
@@ -188,4 +188,51 @@ public void testModifyListWithCache() {
     assertThat(masterDb.getDetails()).containsExactlyInAnyOrder(detail3, detail1);
 
   }
+
+  @Test
+  public void testModifyListWithCache2() {
+    final OmCacheOrderedMaster master = new OmCacheOrderedMaster("Master");
+    final OmCacheOrderedDetail detail1 = new OmCacheOrderedDetail("Detail1");
+    final OmCacheOrderedDetail detail2 = new OmCacheOrderedDetail("Detail2");
+    final OmCacheOrderedDetail detail3 = new OmCacheOrderedDetail("Detail3");
+    DB.save(detail1);
+    DB.save(detail2);
+    DB.save(detail3);
+    master.getDetails().add(detail1);
+    master.getDetails().add(detail2);
+    master.getDetails().add(detail3);
+
+    DB.save(master);
+
+    OmCacheOrderedMaster masterDb = DB.find(OmCacheOrderedMaster.class, master.getId()); // load cache
+    assertThat(masterDb.getDetails()).containsExactly(detail1, detail2, detail3);
+
+    masterDb = DB.find(OmCacheOrderedMaster.class, master.getId());
+    assertThat(masterDb.getDetails()).containsExactly(detail1, detail2, detail3); // hit cache
+
+    // 1 und 2 tauschen
+    masterDb.getDetails().add(0, masterDb.getDetails().remove(1));
+    DB.save(masterDb);
+
+    masterDb.getDetails().remove(2);
+    DB.save(masterDb);
+
+    LoggedSql.start();
+    OmCacheOrderedMaster masterDbNew = DB.find(OmCacheOrderedMaster.class, master.getId());
+    masterDbNew.getDetails().size();
+    assertThat(masterDbNew.getDetails()).containsExactly(detail2, detail1);
+    List<String> sql = LoggedSql.stop();
+    assertThat(sql).hasSize(1).first().asString()
+      .startsWith("select t0.id, t1.id, t1.name, t1.version, t1.sort_order, t1.master_id from om_cache_ordered_master t0 left join om_cache_ordered_detail t1 on t1.master_id = t0.id where t0.id = ? order by t1.sort_order;");
```

Comment:
Note: if the master bean is in cache, we take a different code path (defaultBeanLoader) instead of LoadManyRequest and this produces different queries.
IMHO this query is more complex, as it queries two tables.
In the refactored version (See [here](https://github.com/ebean-orm/ebean/pull/3486/files#diff-860cba4c5ae07ba9d4643752bb11ec9f2043a11f4604eec055ca3dcafaa9028eR229) we got the same query